File: Programmers/내일은 코딩테스트 with 파이썬/파트 12, 그리디 알고리즘/체육복.py
def solution(n, lost, reserve):
    answer = 0
    lost.sort()
    reserve.sort()
    
    for i in reserve[:]:
        # 도난당한 학생 중 여벌 체육복이 있는 경우 제외
        if i in lost:
            reserve.remove(i)
            lost.remove(i)

    for i in reserve:
        if i-1 in lost:
            lost.remove(i-1)
        elif i+1 in lost:
            lost.remove(i+1)
                
    answer = n - len(lost)
    return answer

File: 프로그래머스/1/42862. 체육복/체육복.cs
// 14:36 ~ 15:06

using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = 0;

        List<int> lost_list = lost.ToList();
        List<int> reserve_list = reserve.ToList();
        
        // 여벌 체육복을 가져온 학생이 체육복을 도난당한 경우 미리 처리
        for(int i=1; i<n+1; i++){
            if(lost_list.Contains(i) && reserve_list.Contains(i)){
                lost_list.Remove(i);
                reserve_list.Remove(i);
            }
        }
        
        for(int i=1; i<n+1; i++){
            // i번 학생이 체육복을 도난당한 사람일 때
            if(lost_list.Contains(i)){
                
                // 순수하게 도난당한 경우에 앞 뒤 사람에게 빌릴 수 있는 경우
                if(reserve_list.Contains(i-1)){
                    reserve_list.Remove(i-1);
                    answer ++;
                }
                else if(reserve_list.Contains(i+1)){
                    reserve_list.Remove(i+1);
                    answer ++;
                }  
                
                //Console.WriteLine(i);
            }
            else
                answer ++;
        }
        // Console.WriteLine($"도난당한 사람 중 체육복 빌린 사람 수 {answer}");
        return answer;
    }
}